smart-contracts/wasm-transform/src/artifact_output.rs

```diff
@@ -67,9 +69,22 @@ impl Output for InstantiatedGlobals {
     fn output(&self, out: &mut impl Write) -> OutResult<()> { self.inits.output(out) }
 }
 
+impl Output for ArtifactVersion {
+    fn output(&self, out: &mut impl Write) -> OutResult<()> {
+        match self {
+            ArtifactVersion::V1 => 1u16.output(out),
+        }
+    }
+}
+
 impl<ImportFunc: Output, CompiledCode: RunnableCode> Output for Artifact<ImportFunc, CompiledCode> {
     fn output(&self, out: &mut impl Write) -> OutResult<()> {
-        self.imports.output(out)?;
+        self.version.output(out)?;
```

Comment:
Since the old artifacts have no version marker, but the version marker is used to distinguish old and new artifacts, doesn't this mean that an old artifact with exactly 1 import could be mistaken for a new one?

Reply:
Yes this should be 255 for the version which is not possible as the number (since there are not 127+ allowed imports, and duplicates are disallowed).